### ISSUE TITLE
Error parsing RML mapping over GTFS-Madrid-Bench

### ISSUE THREAD

**Issue by dachafra** [carml/carml#105]
We are trying to generate the RDF based on the attachment data and mappings and we are obtaining the following error when the mapping is being parsed:
```
Exception in thread "main" java.lang.StackOverflowError
	at com.taxonic.carml.rdf_mapper.impl.CarmlMapper.doSingleTypeConcreteClassMapping(CarmlMapper.java:244)
	at com.taxonic.carml.rdf_mapper.impl.CarmlMapper.map(CarmlMapper.java:85)
	at com.taxonic.carml.rdf_mapper.impl.ComplexValueTransformer.transform(ComplexValueTransformer.java:86)
	at com.taxonic.carml.rdf_mapper.impl.IterablePropertyValueMapper.lambda$map$0(IterablePropertyValueMapper.java:37)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.taxonic.carml.rdf_mapper.impl.IterablePropertyValueMapper.map(IterablePropertyValueMapper.java:38)
	at com.taxonic.carml.rdf_mapper.impl.CarmlMapper$2.handle(CarmlMapper.java:516)
	at com.taxonic.carml.rdf_mapper.impl.CarmlMapper.lambda$doSingleTypeConcreteClassMapping$16(CarmlMapper.java:245)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```
The mapping and data runs in other RML parsers (i.e., RMLMapper, RocketRML and SDM-RDFizer)

[gtfs-data.zip](https://github.com/carml/carml/files/6054883/gtfs-data.zip)

**Comment by pmaria** [comment]
@dachafra should be fixed with v0.3.2.

**Comment by dachafra** [comment]
thanks @pmaria, we will include the performance results from CARML if the paper submitted to the KGCW is accepted.

cc @jatoledo @ArenasGuerreroJulian
